### ISSUE TITLE
Install RTL Plugin "Locally"

### ISSUE THREAD

**Issue by neodescis** [maplibre/maplibre-gl-js#3215]
## User Story
Currently the RTL text plugin is provided to MapLibre as a URL. Depending on the environment, this can be cumbersome. It would be much more convenient if the plugin could simply be installed into a project with npm and provided directly.

## Rationale
Many of my projects run in a disconnected environment. Downloading the RTL plugin on-the-fly from a CDN is not an option. One possible solution is to serve the file locally as part of the web application, but this results in an extra HTTP request, and can be less than obvious to get working in some build environments. In Angular, for example, the file needs to be included in the application's "assets" for this to work. This is further complicated if building an Angular library that provides a map component that uses MapLibre, as the consuming application must then have a similar "assets" setup as well. Another option is to copy the file into a string and create a blob URL from that. This is the solution I'm using presently, but it is less than ideal for other (hopefully obvious) reasons. It would be nice to at least have the option of including the plugin directly from an import and using it from there.

## Impact
In my opinion, the current method of providing the RTL plugin is convoluted. Allowing it to be provided directly would make it much easier to drop in the RTL plugin.

**Comment by wipfli** [comment]
Sounds like a good idea to me @neodescis. If this should require creating a new repo in the MapLibre github organization, then the main question for me would be if you would be willing to maintain it?

**Comment by HarelM** [comment]
This is generally possible I believe, I'm doing it already:
https://github.com/IsraelHikingMap/Site/blob/645b0544889446f35de97a0fb30a89c60ec9c4aa/IsraelHiking.Web/src/application/components/map/main-map.component.ts#L113

**Comment by neodescis** [comment]
> This is generally possible I believe, I'm doing it already: https://github.com/IsraelHikingMap/Site/blob/645b0544889446f35de97a0fb30a89c60ec9c4aa/IsraelHiking.Web/src/application/components/map/main-map.component.ts#L113

Yes, this is one of the alternatives I mentioned above. For that solution to work, you have to deliver the .js file as part of the consuming application. There's a glob in angular.json' assets section to do so.

**Comment by neodescis** [comment]
> Sounds like a good idea to me @neodescis. If this should require creating a new repo in the MapLibre github organization, then the main question for me would be if you would be willing to maintain it?

I hope that a fork would not be needed! I know very little about how the plugin itself actually works. Ultimately I will need to play with it to see what's possible.

**Comment by HarelM** [comment]
I must have missed read the description, sorry.
I would recommend allowing to place the content of the exported variable (assuming this is what the plugin does) and not just the url when setting rtl plugin.
It doesn't sound too complicated, hopefully...
I haven't looked through the code to fully understand what it takes to implement this.
Sounds like a good simplification in general.

**Comment by HarelM** [comment]
I'm working on the communication layer between the workers and the main thread now, and there's a full section to help with the RTL plugin, so stating that this is a plugin is a bit of an outreach...
But it allows reducing bundle size, which I guess is the main reason around it...

**Comment by msbarry** [comment]
In a webpack/create-react-app I just switched to

```js
import { setRTLTextPlugin } from 'maplibre-gl';
const mapboxGlRtlUrl = new URL(
  '@mapbox/mapbox-gl-rtl-text/mapbox-gl-rtl-text.min.js',
  import.meta.url
);
setRTLTextPlugin(mapboxGlRtlUrl, () => {}, true);
```

**Comment by neodescis** [comment]
Unfortunately for me, the Angular overlords have decided to make that impossible by disabling webpack's import.meta.url parsing:

https://github.com/angular/angular-cli/issues/24617

Maybe I'll try again when Angular esbuild builder is something I can switch to.
